interop/lib/interop/client.ex

```diff
@@ -191,19 +185,31 @@ defmodule Interop.Client do
         payload: payload(271_828)
       )
 
-    {:ok, res_enum, %{headers: new_headers}} =
+    {headers, data, trailers} =
       ch
-      |> Grpc.Testing.TestService.Stub.full_duplex_call(metadata: metadata)
+      |> Grpc.Testing.TestService.Stub.full_duplex_call(metadata: metadata, return_headers: true)
```

Comment:
I'm not sure if we should add `return_headers` here. Perhaps we should keep a version that has `return_headers` and one that does not.

Reply:
About this one. When using gun adapter, the adapter always returns the headers. :/

I could do as you asked, tho, I might have to write a match case for gun and another for mint. WDYT?

Reply:
Why do we have the `return_headers: option` then?
Is it new to the Mint adapter?

If so, I think it's important to add it on a separate, specific test.

Reply:
That option is not new, its present on the original implementation. 

On gun adapter it's used to know if the trailer headers should be returned. 
On mint if the flag is off I don't return both (the response headers and the trailer headers)

Reply:
I'll work on have a test of this one tho

Reply:
Hey @polvalente! 
I've changed this to behave exactly as Gun adapter. Aka: we don't need that option anymore. Can I mark this one as solved? I'm asking that, cause I have an possible refactor in mind - but it's for after this PR

Tho, I found out that that the `return_headers` option is kinda messy. Basically we have two  [1](https://github.com/elixir-grpc/grpc/blob/master/lib/grpc/stub.ex#L236) [2](https://github.com/elixir-grpc/grpc/blob/master/lib/grpc/stub.ex#L367) places where we can pass that each one is used to return headers for (1) unary / client stream (2) bidirectional_stream / server stream

But the (2) also changes the behavior for trailers. IMO we should have only one of them defining this behavior.

Reply:
Cool! We can discuss improvements on a separate issue/PR for sure!

Regarding the `return_headers` option, it seems that the only problem might be the naming. Trailers are returned in both places whenever the option is activated.

Also, I think that [1](https://github.com/elixir-grpc/grpc/blob/master/lib/grpc/stub.ex#L236) shouldn't be a doc-exposed function anyway. Perhaps turn that doc to a comment and set `@doc false` for the `def call` function. It's only public because `use GRPC.Stub` injects a call to it.

Reply:
yeah! Agreed! We can include the option renaming when addressing these changes

Reply:
I'll do the doc change as you suggested